src/App.tsx

```diff
@@ -15,36 +15,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SOTR_BY_ALPHABET: string = 'alphabet';
+const SOTR_BY_LENGTH: string = 'length';
+
+function SortGoodsFrom(
+  originArray: string[],
+  reverse: boolean,
+  sortBy: string,
+) {
+  let mutateArray: string[] = [...originArray];
+
+  switch (sortBy) {
+    case SOTR_BY_ALPHABET:
+      mutateArray = mutateArray.sort((a, b) => a.localeCompare(b));
+      break;
+    case SOTR_BY_LENGTH:
+      mutateArray = mutateArray.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
+
+  if (reverse) {
+    mutateArray.reverse();
+  }
+
+  return mutateArray;
+}
+
 export const App: React.FC = () => {
+  const [isReverse, setIsReverse] = useState(false);
+  const [sort, setSort] = useState('');
+
+  const visibleGoods: string[] = SortGoodsFrom(
+    goodsFromServer,
+    isReverse,
+    sort,
+  );
+
+  function Reset() {
+    setIsReverse(false);
+    setSort('');
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-danger ${sort === SOTR_BY_ALPHABET ? null : 'is-light'}`}
+          onClick={() => setSort(SOTR_BY_ALPHABET)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-danger ${sort === SOTR_BY_LENGTH ? null : 'is-light'}`}
+          onClick={() => setSort(SOTR_BY_LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button is-danger ${isReverse ? null : 'is-light'}`}
+          onClick={() =>
+            !isReverse ? setIsReverse(true) : setIsReverse(false)
```

Comment:
```suggestion
            setIsReversed(prevState => !prevState)
```
Could be simplified, and when you set state depending on the previous one use functional `setState`